### DISCUSSION TITLE
Suggestion: Decorative Path Additions

### DISCUSSION THREAD

**Discussion by aemp222** [tubbo/openrct2-benchwarmer#81] [Ideas]
Add lamps into the benchwarmer plugin so that it places all three sequentially. I always place them that way in all my parks but it takes a lot of time. Especially since the huge parks take forever to cover with benches, bins, and lamps. With the lamps, it would be phenomenal. Just a humble suggestion

**Comment by ghost** [comment]
+1

Would love this too. I know that lamps do nothing gameplay wise but I always place them down regardless as it makes everything look nice.

Also maybe add an option to fill all queue lines with Queue TVs.

**Comment by tubbo** [comment]
Hey @aemp222 and @BardBun, thanks for the suggestion! I'm glad that y'all have been finding this plugin useful. 

The main goal here is to automate a very tedious task in order to help one progress in the game without developing carpal tunnel syndrome. I chose to leave lamps and other path additions out of this plugin in order to better focus on the utilitarian aspects of it rather than the decorative ones. The problem of visually choosing which object to place down, in addition to the need for multiple themes in the same park (I don't necessarily want the "space lamp" in my wild west section...), is a bit challenging, and since it doesn't really have to do with the goal of the plugin, there's not much motivation for me to work on it.

If anyone is willing to take on this feature, I'll be happy to review and merge it in. Queue TVs definitely do improve guest happiness while waiting in longer lines, so that would be a pretty cool thing to add into the plugin. I suppose the issue I have with lamps is that the plugin is (by design) an "all or nothing" approach, and therefore isn't well-suited for decorative elements rather than utilitarian ones. Perhaps a different plugin that compliments this one would be useful, so one could build decorative path elements like lamps in a selected area of tiles, rather than the whole park.

**Comment by ExileFox** [comment]
>  (I don't necessarily want the "space lamp" in my wild west section...)

here is a couple of ideas on how to circumvent that:
1. make use of a selection area, kinda like the landscape/patrol tool for decorative items.
2. let the user associate certain decorative items with certain path styles

**Comment by tubbo** [comment]
@Ryder17z I've been thinking about this as well especially when playing with the OpenRCT2 beta for the massive new `.park` format. With such massive tile spaces, you're bound to want to use different types of benches/bins for different sections of the park. Or, if you're making scenery like roads, you don't want benches & bins next to the road because that doesn't make any sense (when was the last time you saw a bench on the shoulder of a highway?). So I think I'm going to mark this for the v1.0.0 milestone because it seems relatively useful, but definitely changes how the entire plugin works.

**Comment by tubbo** [comment]
moved to discussions
